### ISSUE TITLE
Accessing to a value of a cell which refer other formula could be long.

### ISSUE THREAD

**Issue by Tarboeuf** [ClosedXML/ClosedXML#662]
## Read and complete the full issue template

- [x] Bug

**Version of ClosedXML**

e.g. 0.90.0.0 head of repo

**What is the current behavior?**

When i want to access to the value of the cell A45, the result is returned after a long moment.
The method GetValue of XlCalcEngine could be called over 2M times.

**Did this work in previous versions of our tool?  Which versions?**

Don't know

**Code to reproduce problem:**
```c#
public void Main()
{
            using (var wb = new XLWorkbook(@"D:\test.xlsx"))
            {
                var ws = wb.Worksheets.First();
                var value = ws.Cell("A45").Value;
               
                Console.WriteLine(value);
            }
}
```
- [x ] I attached a sample spreadsheet.  
[test.xlsx](https://github.com/ClosedXML/ClosedXML/files/1652199/test.xlsx)

**Comment by igitur** [comment]
@vbjay I'm pedantic too, but that's maybe taking it a bit too far? ;-)

**Comment by igitur** [comment]
@Tarboeuf This is a known issue. Formula results aren't cached in any way, and this kind of deep recursion will lead to performance issues. There's no easy away around this.

**Comment by Tarboeuf** [comment]
@igitur, if it's a known issue, I won't bother you with this one, 
And I am pretty sure that's not an easy refactoring, thanks for your answer.

**Comment by igitur** [comment]
Fixed in #679
